File: XMLCheck/Handler/CheckRulesXML01/Xml1RuleBase.cs
using XMLCheck.Models;
using XMLCheck.Services;

namespace XMLCheck.Handler.CheckRulesXML1
{
    public abstract class Xml1RuleBase : IXml1Rule
    {
        public abstract string Key { get; }

        public abstract ErrorDetails? Check(Xml1Model model, HoSoContext context);


        protected ErrorDetails Error(string message)
        {
            return new ErrorDetails
            {
                key = Key,
                message = message
            };
        }
    }

}

File: XMLValidator_v2/src/parsers/rules/xml1/xml1RuleBase.ts
import { Xml1Model } from '../../../interfaces/xml-models.interface';
import { HoSoContext } from '../../../services/hoso-context';
import { ErrorDetails } from '../../../dto/xml-check.dto';

export interface IXml1Rule {
  readonly key: string;
  check(model: Xml1Model, context: HoSoContext): ErrorDetails | null;
}

export abstract class Xml1RuleBase implements IXml1Rule {
  public abstract get key(): string;
  public abstract check(model: Xml1Model, context: HoSoContext): ErrorDetails | null;

  protected error(message: string): ErrorDetails {
    return {
      key: this.key,
      message,
    };
  }
}
